### PR TITLE
[Federation][init-09] Implement unit tests for `kubefed init`.

### PR DESCRIPTION
**Pull request by madhusudancs**
Please review only the last commit here. This is based on PR #35866 which will be reviewed independently.

Design Doc: PR #34484

cc @kubernetes/sig-cluster-federation @quinton-hoole

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35867)

### CODE REVIEW

File: federation/pkg/kubefed/init/init_test.go [hunk 1]
```diff
@@ -288,6 +364,394 @@ func TestCertsHTTPS(t *testing.T) {
...
+				case credSecretName:
+					want = credSecret
+				case cmKubeconfigSecretName:
+					want = cmKubeconfigSecret
+				}
+				if !api.Semantic.DeepEqual(got, want) {
+					return nil, fmt.Errorf("Unexpected secret object\n\tDiff: %s", diff.ObjectGoPrintDiff(got, want))
+				}
+				return &http.Response{StatusCode: http.StatusCreated, Header: kubefedtesting.DefaultHeader(), Body: kubefedtesting.ObjBody(codec, &want)}, nil
+			case p == "/api/v1/namespaces/federation-system/persistentvolumeclaims" && m == http.MethodPost:
```
Comment (nikhiljindal) [inline federation/pkg/kubefed/init/init_test.go]: You made "federation-system" a const in another PR. Use that?

File: federation/pkg/kubefed/init/init_test.go [hunk 2]
```diff
@@ -288,6 +364,394 @@ func TestCertsHTTPS(t *testing.T) {
...
+				if err != nil {
+					return nil, err
+				}
+				switch got.Name {
+				case svcName:
+					want = apiserver
+				case cmName:
+					want = cm
+				}
+				if !api.Semantic.DeepEqual(got.Spec, want.Spec) {
```
Comment (nikhiljindal) [inline federation/pkg/kubefed/init/init_test.go]: You are verifying only spec here. Do you want the complete cm and apiserver deployment definitions? Only define Spec?

**Comment by nikhiljindal** [comment]
LGTM with 2 suggestions.
Feel free to self apply the label

**Comment by madhusudancs** [comment]
Addressed a comment. Adding LGTM label.

**Pull request merged**